src/server/game/Quests/QuestDef.cpp

```diff
@@ -252,15 +252,19 @@ int32 Quest::GetRewOrReqMoney(uint8 playerLevel) const
         }
     }
 
-    return static_cast<int32>(rewardedMoney * sWorld->getRate(RATE_REWARD_BONUS_MONEY));
+    return static_cast<int32>(rewardedMoney * sWorld->getRate(RATE_REWARD_QUEST_MONEY));
 }
 
 uint32 Quest::GetRewMoneyMaxLevel() const
 {
+    uint32 rewMoney = 0;
+
     if (HasFlag(QUEST_FLAGS_NO_MONEY_FROM_XP))
-        return 0;
+        return rewMoney;
+
+    rewMoney = (XPValue(sWorld->getIntConfig(CONFIG_MAX_PLAYER_LEVEL)) * (6 * COPPER));
```

Comment:
This (the copper multiplication) was done for readability, can be removed if need be.